sigauth/signature.go

```diff
@@ -154,16 +158,40 @@ func (s *SignatureAuth) CheckRequestServiceSignature(r *Request, requiredService
 		return "", err
 	}
 
-	if requiredServiceIDs != nil && !authutils.ContainsString(requiredServiceIDs, sigAuthHeader.KeyID) {
-		return "", fmt.Errorf("request signer (%s) is not one of the required services %v", sigAuthHeader.KeyID, requiredServiceIDs)
+	var serviceReg *authservice.ServiceReg
+	var fingerprint string
+	found := false
+	if requiredServiceIDs == nil {
+		requiredServiceIDs = s.serviceRegManager.SubscribedServices()
+	}
+
+	for _, serviceID := range requiredServiceIDs {
+		serviceReg, err = s.serviceRegManager.GetServiceRegWithPubKey(serviceID)
+		if err != nil {
+			return "", fmt.Errorf("failed to retrieve service pub key: %v", err)
+		}
+
+		fingerprint, err = authutils.GetKeyFingerprint(&s.serviceKey.PublicKey)
```

Comment:
The `ServiceReg.PubKey` should actually have this key ID computed and cached in the `Kid` field. Feel free to update naming for consistency, but we should avoid the redoing fingerprint operation unnecessarily as it is not super cheap.